### ISSUE TITLE
Image AspectFill not working

### ISSUE THREAD

**Issue by dirivero** [dotnet/maui#6378]
### Description

AspectFill doesn't work well in Android or iOS. Here is a base scenario:

```
<Grid HeightRequest="300" VerticalOptions="Start" BackgroundColor="Red" Padding="10">
    <Image 
        Source="maui.jpg" 
        Aspect="AspectFill"
     />
</Grid>
```

This is how it looks in Xamarin as reference:

![Simulator Screen Shot - iPhone 13 mini - 2022-04-21 at 10 29 11 (1)](https://user-images.githubusercontent.com/7670773/164469751-cc28c312-a517-4e4e-b75f-2c86b3fad1ab.png)

**Android - doesn't honor the position and doesn't clip**

![image](https://user-images.githubusercontent.com/7670773/164466771-0838e0b0-4f94-4075-bc04-f3c42290a192.png)

**iOS - doesn't clip**

![Simulator Screen Shot - iPhone 13 - 2022-04-21 at 10 38 03](https://user-images.githubusercontent.com/7670773/164470405-72f2591c-d429-4593-993a-2fff77dc6c8c.png)

This is a similar scenario but with the image having a fixed height:

```
<Grid HeightRequest="300" VerticalOptions="Start" BackgroundColor="Red" Padding="10">
    <Image
        VerticalOptions="Center"
        HeightRequest="100"
        Source="maui.jpg"
        Aspect="AspectFill"
    />
</Grid>
```

This is how it looks in Xamarin as reference:

![Simulator Screen Shot - iPhone 13 mini - 2022-04-21 at 10 44 39](https://user-images.githubusercontent.com/7670773/164471581-07cd8f97-21ec-4dc7-ad92-2573afc25c4a.png)

**Android - doesn't honor the position and doesn't clip**

![image](https://user-images.githubusercontent.com/7670773/164466485-e5ceec62-5b8d-4d25-b428-466386adc6d7.png)

**iOS - doesn't clip**

Same screenshot as before.

This is related to [issue 2741](https://github.com/dotnet/maui/issues/2741) but is specific for AspectFill and covers 2 very common use of images.

### Steps to Reproduce

Is reproduced with the XAML snippets above.

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

All

**Comment by kristinx0211** [comment]
can be reproduced on android 11 with above code.

**Comment by Enbi-Zhang** [comment]
Verified this issue with Visual Studio Enterprise 17.6.0 Preview 7.0. Can repro on android platform with above project. 
[6378.zip](https://github.com/dotnet/maui/files/11515333/6378.zip)
![6378](https://github.com/dotnet/maui/assets/95854506/c2277a96-9b3d-49e6-9622-53382326ca8b)

**Comment by donatellijl** [comment]
This is big issue for migrating to MAUI. I upgraded my Xamarin.Forms app and my layouts broke because the images are filling in strange ways.

**Comment by jknaudt21** [comment]
iOS will be fixed by: #17120

I'll start looking into Android

**Comment by jknaudt21** [comment]
@samhouts  I'm tempted to close this one unless I'm misunderstanding the expected behavior.  Both the iOS and Android issues seem to have been solved. 

Using the code below, I get the following results testing on main:
```
    <Grid HeightRequest="300" VerticalOptions="Start" BackgroundColor="Red" Padding="10">
        <Image Source="dotnet_bot.png" Aspect="AspectFill"/>
    </Grid>
```


| iOS | MacCatalyst | Android | 
| -- | -- | -- |
| <img width="430" alt="image" src="https://github.com/dotnet/maui/assets/32918747/5ac9799f-ab84-411c-b384-ea2f9b841ac8"> | <img width="272" alt="image" src="https://github.com/dotnet/maui/assets/32918747/623aef4a-6f23-4ff9-b4e0-fad465ce3259"> |  <img width="354" alt="image" src="https://github.com/dotnet/maui/assets/32918747/b1f0c539-ef45-4f32-a80c-0744aad47674"> |
  
In all cases the height is respected and the image expands to fill its container. The image clips itself in the right places too.
